### ISSUE TITLE
大佬麻烦看一下There appear to be 6 leaked semaphores to clean up at shutdown

### ISSUE THREAD

**Issue by 54hg0220** [yl305237731/flexible-yolov5#40]
首先膜拜下大佬，我在colab上用自己的数据集训练显示以下错误
fatal: ambiguous argument 'main..origin/master': unknown revision or path not in the working tree.
Use '--' to separate paths from revisions, like this:
'git <command> [<revision>...] -- [<file>...]'
github: Command 'git rev-list main..origin/master --count' returned non-zero exit status 128.
YOLOv5 c5b7925 torch 1.9.0+cu102 CPU

Namespace(adam=False, batch_size=16, bucket='', cache_images=False, cfg='configs/model_efficientnet.yaml', data='configs/data.yaml', device='', epochs=100, evolve=False, exist_ok=False, global_rank=-1, hyp='configs/hyp.scratch.yaml', image_weights=False, img_size=[640, 640], linear_lr=False, local_rank=-1, log_artifacts=False, log_imgs=16, multi_scale=False, name='exp', noautoanchor=False, nosave=False, notest=False, project='runs/train', quad=False, rect=False, resume=False, save_dir='runs/train/exp5', single_cls=False, sync_bn=False, total_batch_size=16, weights='', workers=8, world_size=1)
Start Tensorboard with "tensorboard --logdir runs/train", view at http://localhost:6006/
2021-07-27 07:36:17.237677: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
hyperparameters: lr0=0.01, lrf=0.2, momentum=0.937, weight_decay=0.0005, warmup_epochs=3.0, warmup_momentum=0.8, warmup_bias_lr=0.1, box=0.05, cls=0.5, cls_pw=1.0, obj=1.0, obj_pw=1.0, iou_t=0.2, anchor_t=4.0, fl_gamma=0.0, hsv_h=0.015, hsv_s=0.7, hsv_v=0.4, degrees=0.0, translate=0.1, scale=0.5, shear=0.0, perspective=0.0, flipud=0.0, fliplr=0.5, mosaic=1.0, mixup=0.0
./od/models/model.py:22: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  model_config = yaml.load(open(model_config, 'r'))
FPN input channel size: C3 88, C4 248, C5 2816
FPN output channel size: P3 344, P4 256, P5 2816
PAN input channel size: P3 344, P4 256, P5 2816
PAN output channel size: PP3 256, PP4 512, PP5 1024
Scaled weight_decay = 0.0005
Optimizer groups: 345 .bias, 345 conv.weight, 220 other
wandb: (1) Create a W&B account
wandb: (2) Use an existing W&B account
wandb: (3) Don't visualize my results
wandb: Enter your choice: 3
wandb: You chose 'Don't visualize my results'
2021-07-27 07:36:39.867593: I tensorflow/stream_executor/platform/default/dso_loader.cc:53] Successfully opened dynamic library libcudart.so.11.0
wandb: W&B syncing is set to `offline` in this directory.  Run `wandb online` or set WANDB_MODE=online to enable cloud syncing.
train: Scanning '/content/Mydata/labels/train' for images and labels... 4302 found, 2 missing, 0 empty, 0 corrupted: 100% 4304/4304 [00:04<00:00, 927.71it/s]
train: New cache created: /content/Mydata/labels/train.cache
val: Scanning '/content/Mydata/labels/valid' for images and labels... 1076 found, 0 missing, 0 empty, 0 corrupted: 100% 1076/1076 [00:01<00:00, 942.23it/s]
val: New cache created: /content/Mydata/labels/valid.cache
Plotting labels... 

autoanchor: Analyzing anchors... anchors/target = 4.52, Best Possible Recall (BPR) = 1.0000
Image sizes 640 train, 640 test
Using 2 dataloader workers
Logging results to runs/train/exp5
Starting training for 100 epochs...

     Epoch   gpu_mem       box       obj       cls     total   targets  img_size
  0% 0/269 [00:00<?, ?it/s]tcmalloc: large alloc 1258291200 bytes == 0x55c498936000 @  0x7f72d340db6b 0x7f72d342d379 0x7f726a07526e 0x7f726a0769e2 0x7f72adec39f8 0x7f72adead359 0x7f72adeba1bf 0x7f72adebb5a7 0x7f72adeb5dbb 0x7f72adeb64c7 0x7f72ae51bc62 0x7f72ae36d57b 0x7f72af9b8c01 0x7f72af9b9392 0x7f72adfe156d 0x7f72ada78518 0x7f72ae58e2ba 0x7f72adfdba7b 0x7f72ada711db 0x7f72ae58e21a 0x7f72adfd9fc5 0x7f72ada70daa 0x7f72ae58e552 0x7f72adfe087d 0x7f72c0975026 0x55c25a09b010 0x55c25a09ada0 0x55c25a10f2f9 0x55c25a09cb99 0x55c25a09d1f1 0x55c25a10c318
tcmalloc: large alloc 1258291200 bytes == 0x55c4e3936000 @  0x7f72d340db6b 0x7f72d342d379 0x7f726a07526e 0x7f726a0769e2 0x7f72ad8e0b49 0x7f72ad8e1897 0x7f72adcbdd89 0x7f72ae422b9a 0x7f72ae405cbe 0x7f72ae00aa05 0x7f72adece86a 0x7f72adeb6594 0x7f72ae51bc62 0x7f72ae36d57b 0x7f72af9b8c01 0x7f72af9b9392 0x7f72adfe156d 0x7f72ada78518 0x7f72ae58e2ba 0x7f72adfdba7b 0x7f72ada711db 0x7f72ae58e21a 0x7f72adfd9fc5 0x7f72ada70daa 0x7f72ae58e552 0x7f72adfe087d 0x7f72c0975026 0x55c25a09b010 0x55c25a09ada0 0x55c25a10f2f9 0x55c25a09cb99
/usr/lib/python3.7/multiprocessing/semaphore_tracker.py:144: UserWarning: semaphore_tracker: There appear to be 6 leaked semaphores to clean up at shutdown
  len(cache))
^C

**Comment by yl305237731** [comment]
@54hg0220 efficientnet config 里面默认写的 b8， b8模型太大了，换一个小一点的 b0,b1,b2啊这些

**Comment by 54hg0220** [comment]
> @54hg0220 efficientnet config 里面默认写的 b8， b8模型太大了，换一个小一点的 b0,b1,b2啊这些
感谢大佬回复，我把config里面的b8模型改成了b0. 然后报错变成了这个 。
Epoch   gpu_mem       box       obj       cls     total   targets  img_size
  0% 0/269 [00:00<?, ?it/s]/usr/lib/python3.7/multiprocessing/semaphore_tracker.py:144: UserWarning: semaphore_tracker: There appear to be 6 leaked semaphores to clean up at shutdown
  len(cache))
^C
并且我试了一下mobileNet， 训练一个epoch就要三个小时😂
![image](https://user-images.githubusercontent.com/79192628/127134170-7811e2f8-debc-4655-a42d-85d6ca3fceb9.png)

**Comment by yl305237731** [comment]
@54hg0220 gpu_mem 为啥是0 啊

**Comment by 54hg0220** [comment]
> @54hg0220 gpu_mem 为啥是0 啊

对不起，我开了两个colab然后另外那边把gpu给关了。。问题解决了😂。被自己给蠢到了

**Comment by 54hg0220** [comment]
> @54hg0220 gpu_mem 为啥是0 啊

大佬方便推荐几个相关的学习网站或者资料吗，我最近在学习不同的神经网络对于手势分类任务的表现

**Comment by yl305237731** [comment]
@54hg0220 吴恩达或者李宏毅的视频呗
